tests/components/heos/test_media_player.py

```diff
@@ -1051,3 +1052,36 @@ async def test_media_player_unjoin_group(
         blocking=True,
     )
     assert "Failed to ungroup media_player.test_player" in caplog.text
+
+
+async def test_media_player_group_fails_when_entity_removed(
+    hass: HomeAssistant,
+    config_entry,
+    config,
+    controller,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test grouping fails when entity removed."""
+    await setup_platform(hass, config_entry, config)
+
+    # Remove one of the players
+    er.async_get(hass).async_remove("media_player.test_player_2")
```

Comment:
the entity registry can be used as a test fixture: `entity_registry: er.EntityRegistry`

Reply:
Updated!